packages/client/src/runtime/core/engines/data-proxy/utils/getClientVersion.ts

```diff
@@ -31,9 +31,11 @@ async function _getClientVersion(host: string, config: EngineConfig) {
 
   // if it is an integration or dev version, we resolve its dataproxy
   // for this we infer the data proxy version from the engine version
-  if (suffix !== undefined || clientVersion === '0.0.0') {
-    // when we are running in tests, then we are using the mini proxy
-    if (process.env.TEST_DATA_PROXY !== undefined) return '0.0.0'
+  if (suffix !== undefined || clientVersion === '0.0.0' || clientVersion === 'in-memory') {
+    // we use the generated engine version to infer if we're in a test
+    if (config.engineVersion === '0000000000000000000000000000000000000000') {
+      return '0.0.0' // when we are running in tests, we use mini proxy
+    }
```

Comment:
Env vars are not available for the edge client tests, which means it actually tried to resolve the real data proxy url, wasting time on CI (considerably). Now instead we just check if the client engine version is the one from the test setup.